packages/shared/src/components/post/PostContentHeaderMobile.tsx

```diff
@@ -4,46 +4,27 @@ import { Button, ButtonSize, ButtonVariant } from '../buttons/Button';
 import { ArrowIcon } from '../icons';
 import { PostHeaderActions } from './PostHeaderActions';
 import { PostNavigationProps } from './common';
-import { SourceType } from '../../graphql/sources';
+import { isNullOrUndefined } from '../../lib/func';
 
 export function PostContentHeaderMobile({
   post,
   onReadArticle,
 }: Pick<PostNavigationProps, 'onReadArticle' | 'post'>): ReactElement {
   const router = useRouter();
-  const canGoBack = !!globalThis?.window?.history?.length;
-  const routedFromSquad = router?.query?.squad;
-  const squadLink = `/squads/${router.query.squad}`;
-
-  const onBackClick = () => {
-    if (canGoBack) {
-      return router.back();
-    }
-
-    if (routedFromSquad) {
-      return router.push(squadLink);
-    }
-
-    if (post.source.type === SourceType.Squad) {
-      return router.push(post.source.permalink);
-    }
-
-    return null;
-  };
+  const canGoBack =
+    globalThis?.history &&
+    !!globalThis.history.length &&
+    isNullOrUndefined(globalThis.history.state.options.shallow) &&
+    !globalThis.document.referrer; // empty referrer means you are from the same site
 
   return (
     <span className="-mx-4 flex flex-row items-center border-b border-border-subtlest-tertiary px-4 py-2 tablet:hidden">
       <Button
         icon={<ArrowIcon className="-rotate-90" />}
         size={ButtonSize.Small}
         variant={ButtonVariant.Tertiary}
-        onClick={onBackClick}
-        className={
-          !canGoBack &&
-          !routedFromSquad &&
-          post.source.type !== SourceType.Squad &&
-          'invisible'
-        }
+        onClick={router.back}
+        className={!canGoBack && 'invisible'}
```

Comment:
Should we just do hidden rather then invisible?

Reply:
We want to keep the height consistent, so we used invisible. We can use either of the two, just saw some benefits here.